questdb-rs/src/ingress/mod.rs

```diff
@@ -1559,12 +1786,43 @@ impl SenderBuilder {
         C: Into<String>,
         D: Into<String>,
     {
-        self.auth = Some(AuthParams {
+        self.auth = Some(AuthParams::Ecdsa(EcdsaAuthParams {
             key_id: key_id.into(),
             priv_key: priv_key.into(),
             pub_key_x: pub_key_x.into(),
             pub_key_y: pub_key_y.into(),
-        });
+        }));
+        self
+    }
+
+    /// Basic Authentication Parameters for ILP over HTTP.
+    /// For TCP, use [`auth`](SenderBuilder::auth).
+    ///
+    /// For HTTP you can also use [`token_auth`](SenderBuilder::token_auth).
+    #[cfg(feature = "ilp-over-http")]
```

Comment:
why not encapsulate all http related functions in a module and import them via feature cfg?
```
 #[cfg(feature = "ilp-over-http")]
mod http {
    // ...
}
 #[cfg(feature = "ilp-over-http")]
pub use http::*;
```

Reply:
Mostly done this now.
Code's much better now, I was rushing through it.